### ISSUE TITLE
Working alongside thomseddon/traefik-forward-auth:2

### ISSUE THREAD

**Issue by jonwilliams84** [fbonalair/traefik-crowdsec-bouncer#16]
Hi,

I am trying to get this to work alongside thomseddon/traefik-forward-auth:2 to add an additional layer of forward-auth but I cant seem to get it working.

Traefik version 2.6.1

Traefik args:
```
      - "--entrypoints.http.http.middlewares=crowdsec-bouncer@docker"
      - "--entrypoints.https.http.middlewares=crowdsec-bouncer@docker"
 ```

docker-compose.yaml
```
version: '3.8'
services:
  crowdsec:
    image: crowdsecurity/crowdsec:latest
    environment:
      GID: "${GID-1000}"
      COLLECTIONS: "crowdsecurity/linux crowdsecurity/traefik"
    volumes:
      - ./config/acquis.yaml:/etc/crowdsec/acquis.yaml
      - crowdsec-db:/var/lib/crowdsec/data/
      - crowdsec-config:/etc/crowdsec/
      - /ztemp/logs/traefik:/var/log/traefik/:ro
    networks:
      - traefik_public
    restart: unless-stopped
    deploy:
      labels:
        - traefik.enable=false

  bouncer-traefik:
    image: docker.io/fbonalair/traefik-crowdsec-bouncer:latest
    container_name: bouncer-traefik
    environment:
      CROWDSEC_BOUNCER_API_KEY: <my-api-key>
      CROWDSEC_AGENT_HOST: crowdsec:8080
    networks:
      - traefik_public
    depends_on:
      - crowdsec
    restart: unless-stopped
    deploy:
      labels:
        - "traefik.enable=true"
        - "traefik.http.middlewares.crowdsec-bouncer.forwardauth.address=http://bouncer-traefik:8080/api/v1/forwardAuth"
        - "traefik.http.middlewares.crowdsec-bouncer.forwardauth.trustForwardHeader=true"
        - "traefik.http.services.crowdsec-bouncer.loadbalancer.server.port=8080"

networks:
  traefik_public:
    external: true
volumes:
  crowdsec-db:
```

I am seeing the middleware in the traefik dash:
![image](https://user-images.githubusercontent.com/24478629/154060644-b0af13eb-8ac3-48a4-9c00-27973543f746.png)

And if I look at my routers, both the foward-auth and the bouncer is associated:
![image](https://user-images.githubusercontent.com/24478629/154060781-dca06eb9-06f7-449a-9b39-7ef7b6cb6139.png)

![image](https://user-images.githubusercontent.com/24478629/154060828-d4b237f5-c79d-4788-b82b-3342569da586.png)
![image](https://user-images.githubusercontent.com/24478629/154060871-63ad166a-5ac5-4327-8761-b5ce79734458.png)

However the request from blocked IPs DOES NOT get forbidden.

Oddly, I also have a static service+router defined for a bare metal box outside docker and that is now also picking up the bouncer middleware and requests from blocked clients IS being blocked.

I have tried removing the other foward-auth middleware from one of my services so its only got the bouncer associated and its still not rejecting banned clients:

![image](https://user-images.githubusercontent.com/24478629/154061566-70e9a274-6b6e-4a45-8cdd-c099ed6bc10a.png)

Here are some logs from the bouncer:
```
{"level":"info","status":200,"method":"GET","path":"/api/v1/forwardAuth","ip":"172.16.200.3","latency":45.521328,"user_agent":"Mozilla/5.0 (Linux; Android 12; SM-G981B) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Mobile Safari/537.36","time":"2022-02-15T12:26:43Z","message":"Request"}
{"level":"info","status":200,"method":"GET","path":"/api/v1/forwardAuth","ip":"172.16.200.3","latency":77.551311,"user_agent":"Mozilla/5.0 (Linux; Android 12; SM-G981B) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Mobile Safari/537.36","time":"2022-02-15T12:26:43Z","message":"Request"}
{"level":"info","status":200,"method":"GET","path":"/api/v1/forwardAuth","ip":"172.16.200.3","latency":32.077496,"user_agent":"Mozilla/5.0 (Linux; Android 12; SM-G981B) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Mobile Safari/537.36","time":"2022-02-15T12:26:43Z","message":"Request"}
{"level":"info","status":200,"method":"GET","path":"/api/v1/forwardAuth","ip":"172.16.200.3","latency":12.25493,"user_agent":"Mozilla/5.0 (Linux; Android 12; SM-G981B) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Mobile Safari/537.36","time":"2022-02-15T12:26:43Z","message":"Request"}
{"level":"info","status":200,"method":"GET","path":"/api/v1/forwardAuth","ip":"172.16.200.3","latency":26.94213,"user_agent":"Mozilla/5.0 (Linux; Android 12; SM-G981B) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Mobile Safari/537.36","time":"2022-02-15T12:26:43Z","message":"Request"}
{"level":"info","status":200,"method":"GET","path":"/api/v1/forwardAuth","ip":"172.16.200.3","latency":20.995469,"user_agent":"Mozilla/5.0 (Linux; Android 12; SM-G981B) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Mobile Safari/537.36","time":"2022-02-15T12:26:43Z","message":"Request"}
{"level":"info","status":200,"method":"GET","path":"/api/v1/forwardAuth","ip":"172.16.200.3","latency":15.450635,"user_agent":"Mozilla/5.0 (Linux; Android 12; SM-G981B) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Mobile Safari/537.36","time":"2022-02-15T12:26:47Z","message":"Request"}
{"level":"warn","status":403,"method":"GET","path":"/api/v1/forwardAuth","ip":"172.16.200.3","latency":25.430298,"user_agent":"Mozilla/5.0 (Linux; Android 12; SM-G981B) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Mobile Safari/537.36","time":"2022-02-15T12:26:51Z","message":"Request"}
{"level":"warn","status":403,"method":"GET","path":"/api/v1/forwardAuth","ip":"172.16.200.3","latency":12.728283,"user_agent":"Mozilla/5.0 (Linux; Android 12; SM-G981B) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Mobile Safari/537.36","time":"2022-02-15T12:26:52Z","message":"Request"}
```
The bottom two are blocks on a file provider service, the others are docker services.

Any ideas? It's probably just a label/mis-config somewhere but I have tried loads of combinations to get to this stage.

Thanks in advance.

**Comment by jonwilliams84** [comment]
I am an idiot. It was working fine all along. I just remembered that the services that "weren't working" were actually protected by cloudflare proxy, thus the client IP is an edge server not the true client IP! 

Please close this and save me the embarrassment! 

Great work on the bouncer by the way! :-)

**Comment by fbonalair** [comment]
> Great work on the bouncer by the way! :-)

Thanks !



> I am trying to fix similar (or same?) issue. In the logs the ip that gets forwarded is traefiks internal ip address instead of the public forwarded address from `X-Forwarded-For` or `X-Real-IP` headers which are being set by traefik real-ip plugin. Is there any current fix for that or it's not supported yet?
> 
> I guess it's this: #10

Exactly, it should be fixed in version 0.3.4. Feel free to comment on the PR if it doesn't fix your case.

**Comment by modem7** [comment]
@jonwilliams84

Sorry for replying on a closed ticket.

Would you be able to share your full compose file so I can see how you did the forward auth stuff etc?

I'm looking to do similar with authelia etc.

Cheers and apologies again!
